docs/extensions.adoc

```diff
@@ -1047,3 +1047,49 @@ def 'test parameter injection'(MyInjectable myInjectable) {
 }
 ----
 --
+
+[[MockMaker]]
+=== MockMaker Extensions
+
+Spock creates `Mock/Stub/Spy` objects via the `IMockMaker` interface.
+When Spock is creating a mock, it will ask the available mock makers in order of priority, if it can mock the required type.
+
+Every mock maker has a unique ID, which can be used by the users to select a mock maker, if needed.
+
+The following mock maker are built-in, and are selected in this order:
+
+* `java-proxy-mock-maker`: Uses the `java.lang.reflect.Proxy` API to create mocks of interfaces.
```

Comment:
The reason behind this was for somewhat consistent error messages like:

```
Cannot create mock for class java.lang.StringBuilder.
dynamic-proxy-mock-maker: Type is not an interface.
byte-buddy-mock-maker: Cannot mock final classes.
cglib-mock-maker: Cannot mock final classes.
```
The suffix ensures that an error message containing that ID can be linked to the `IMockMaker`.
If e.g. another part will be involved it is more distingishable, e.g.:

```
Cannot create mock for class java.lang.StringBuilder.
dynamic-proxy-mock-maker: Type is not an interface.
objenesis-instatiator: Library not found.
```

Reply:
Couldn't that maybe also be achieved by either not using `mockMaker.getId()` at the two places you use it in message,
but either by using `mockMaker` and having a meaningful `toString()` method which for example returns `"ByteBuddy mock maker"` which actually is the purpose of a proper `toString()` method,
or by using `mockMaker.getId() + " mock maker"` or `"mock maker with ID " + mockMaker.getId()`?

Reply:
I still think that the suffix is better, because it is also API in the configuration object.
Furthermore, I can't rely on the `mockmaker.toString()` to yield good results for 3rd-Party `MockMakers`.
I have the hope by defining the suffx `-mock-maker`, that people defining a mock maker will use similar names as Spock does.

Reply:
Done.